x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/external/request/googleaistudio/GoogleAiStudioCompletionRequestEntity.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.inference.external.request.googleaistudio;
+
+import org.elasticsearch.xcontent.ToXContentObject;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.Objects;
+
+public record GoogleAiStudioCompletionRequestEntity(List<String> input) implements ToXContentObject {
+
+    private static final String CONTENTS_FIELD = "contents";
+
+    private static final String PARTS_FIELD = "parts";
+
+    private static final String TEXT_FIELD = "text";
+
+    private static final String GENERATION_CONFIG_FIELD = "generationConfig";
+
+    private static final String CANDIDATE_COUNT_FIELD = "candidateCount";
+
+    private static final String ROLE_FIELD = "role";
+
+    private static final String ROLE_USER = "user";
+
+    public GoogleAiStudioCompletionRequestEntity {
+        Objects.requireNonNull(input);
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
```

Comment:
Nit: does every field need its own surrounding object?

Reply:
Yes, this is the entity representing the request, which will be send to Google AI Studio, so they expect exactly this structure.